src/client/cockpit-beiboot

```diff
@@ -0,0 +1,312 @@
+#!/usr/bin/env python3
+
+# This file is part of Cockpit.
+#
+# Copyright (C) 2022 Red Hat, Inc.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+import argparse
+import asyncio
+import base64
+import importlib.resources
+import json
+import logging
+import lzma
+import os
+import shlex
+import sys
+from pathlib import Path
+from typing import Iterable, Optional, Sequence
+
+from cockpit import polyfills
+from cockpit._vendor import ferny
+from cockpit._vendor.bei import beipack, bootloader
+from cockpit.bridge import setup_logging
+from cockpit.channel import ChannelRoutingRule
+from cockpit.channels import PackagesChannel
+from cockpit.data import read_cockpit_data_file
+from cockpit.packages import Package, Packages
+from cockpit.peer import Peer
+from cockpit.protocol import CockpitProblem
+from cockpit.router import Router, RoutingRule
+from cockpit.transports import StdioTransport
+
+logger = logging.getLogger(__name__)
+
+
+def ensure_ferny_askpass() -> Path:
+    src_path = importlib.resources.files(ferny.__name__) / 'interaction_client.py'
+    src_data = src_path.read_text()
+
+    xdg_data_home = os.environ.get('XDG_DATA_HOME')
+    if xdg_data_home is None:
+        xdg_data_home = os.path.expanduser('~/.local/share')
```

Comment:
IMHO this belongs into `XDG_CACHE_HOME/cockpit` aka. ~/.cache/cockpit (just in case we are going to cache more stuff in the future?) This isn't precious user data.

Reply:
Not sure I agree.  User data doesn't belong in share anyway, that's more like "stuff that someone might also find in `/usr/share`" (like cockpit packages and so).  This fits more to the template of "something that might have been installed in `/usr`".

I'm also (perhaps excessively) allergic to installing things in `~/.cache` if we depend on their existence, even 1 second later.  In theory, `~/.cache` is fail game to be deleted.

Reply:
My litmus test for this is "does this need to be in your backup". ~/local/share has all kinds of precious user data, from gtimelog's diaries over custom desktop launchers to saved games. It surely also has some crap, but let's please not add to that. This is classic cache material. It doesn't mean that you can expect to delete the cache underneath running programs and expect it to work, just that you can restart your session without any data loss.

If you don't like that, I'm also fine with `$XDG_RUNTIME_DIR`, which is perhaps even better wrt. the "never hit the disk" attitude of beiboot?

Reply:
I'd also have taken `XDG_RUNTIME_DIR` as my first pick here, but it has a pretty big disadvantage in that it doesn't have the same path "inside" and "outside" of the flatpak.

Reply:
I let you win... This time :wink: